src/Stratis.Bitcoin.Features.Consensus.Tests/PeerBanningTest.cs

```diff
@@ -0,0 +1,131 @@
+        {
+            string dataDir = Path.Combine("TestData", nameof(PeerBanningTest), nameof(this.GivenANodeIsSyncedAndConnected_WhenAPeerSendsABadBlockAndErrorIsNotBanError_ThePeerIsNotBanned_Async));
+            Directory.CreateDirectory(dataDir);
+
+            TestChainContext context = await TestChainFactory.CreateAsync(Network.RegTest, dataDir);
+            var peer = new IPEndPoint(IPAddress.Parse("1.2.3.4"), context.Network.DefaultPort);
+
+            var connectionManagerBehavior = new ConnectionManagerBehavior(false, context.ConnectionManager, context.LoggerFactory) { Whitelisted = true };
+            var node = new Node();
+            node.Behaviors.Add(connectionManagerBehavior);
+            context.MockReadOnlyNodesCollection.Setup(s => s.FindByEndpoint(It.IsAny<IPEndPoint>())).Returns(node);
+
+            var blocks = await TestChainFactory.MineBlocksAsync(context, 2, new Key().ScriptPubKey);
+            // create a new block that breaks consensus.
+            var block = blocks.First();
+            block.Header.HashPrevBlock = context.Chain.Tip.HashBlock;
+            await context.Consensus.AcceptBlockAsync(new BlockValidationContext { Block = block, Peer = peer, BanDurationSeconds = BlockValidationContext.BanDurationNoBan });
+
+            Assert.False(context.PeerBanning.IsBanned(peer));
+        }
+
+        [Fact]
+        public async Task GivenANodeIsSyncedAndConnected_WhenAPeerSendsABadBlockAndPeerIsNotWhitelisted_ThePeerGetsBanned_Async()
+        {
+            string dataDir = Path.Combine("TestData", nameof(PeerBanningTest), nameof(this.GivenANodeIsSyncedAndConnected_WhenAPeerSendsABadBlockAndPeerIsNotWhitelisted_ThePeerGetsBanned_Async));
+            Directory.CreateDirectory(dataDir);
+
+            TestChainContext context = await TestChainFactory.CreateAsync(Network.RegTest, dataDir);
+            var peer = new IPEndPoint(IPAddress.Parse("1.2.3.4"), context.Network.DefaultPort);
+
+            var connectionManagerBehavior = new ConnectionManagerBehavior(false, context.ConnectionManager, context.LoggerFactory);
+            var node = new Node();
+            node.Behaviors.Add(connectionManagerBehavior);
+            context.MockReadOnlyNodesCollection.Setup(s => s.FindByEndpoint(It.IsAny<IPEndPoint>())).Returns(node);
+
+            var blocks = await TestChainFactory.MineBlocksAsync(context, 2, new Key().ScriptPubKey);
+            // create a new block that breaks consensus.
+            var block = blocks.First();
+            block.Header.HashPrevBlock = context.Chain.Tip.HashBlock;
+            await context.Consensus.AcceptBlockAsync(new BlockValidationContext { Block = block, Peer = peer, BanDurationSeconds = 1}); // ban for 1 sec
+            
+            // wait 1 sec for ban to expire.
+            Thread.Sleep(1000);
+
+            Assert.False(context.PeerBanning.IsBanned(peer));
```

Comment:
should this test body not be in previous test? 
your test method names do not match what the test is doing actually, check last 3 tests here